src/app/services/sync/item.service.ts

```diff
@@ -106,4 +106,11 @@ export class ItemService {
     });
   }
 
+  filterItems(items, searchTerm) {
+    return items.filter((item) => {
```

Comment:
By filtering I meant getting results from the server by the filter and using cache when offline.